extensions/dataloading/src/main/java/org/eclipse/dataspaceconnector/dataloading/AssetLoader.java

```diff
@@ -22,4 +22,13 @@
     String FEATURE = "edc:asset:assetindex:loader";
 
     void accept(Asset asset, DataAddress dataAddress);
+
+    /**
+     * Deletes an asset.
+     *
+     * @param assetId Id of the asset to be deleted.
+     * @return Deleted Asset or null if asset did not exist.
+     */
```

Comment:
minor nit: state that an `EdcPersistenceException` will be thrown in case an error happens